radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/ExecutionContext.java

```diff
@@ -101,4 +153,12 @@ public void verifyPermissionLevel(PermissionLevel requiredLevel) throws SignedSy
 			throw new SignedSystemException();
 		}
 	}
+
+	public void destroy() throws DefaultedSystemLoanException, ExecutionContextDestroyException {
+		payOffLoan();
+
+		if (!reserve.isEmpty()) {
+			throw new ExecutionContextDestroyException(reserve);
+		}
```

Comment:
It might be worth go get product's feedback on this. Some users may not want their balances ending up with `***.999999123`.

Reply:
You mean not enforcing exact fees?